Sources/SPTDataLoader.m

```diff
@@ -130,6 +132,26 @@ - (void)removeRequest:(SPTDataLoaderRequest *)request
     return cancellationToken;
 }
 
+- (nullable id<SPTDataLoaderCancellationToken>)performRequest:(SPTDataLoaderRequest *)request
+                                               withCompletion:(void (^)(SPTDataLoaderResponse *response))completion
+{
+    SPTDataLoaderRequest *copiedRequest = [request copy];
+    id<SPTDataLoaderCancellationToken> cancellationToken = [self.cancellationTokenFactory createCancellationTokenWithDelegate:self
+                                                                                                                 cancelObject:copiedRequest];
+    copiedRequest.cancellationToken = cancellationToken;
+    @synchronized(self.cancellationTokens) {
+        [self.cancellationTokens addObject:cancellationToken];
+    }
+    @synchronized(self.requests) {
+        [self.requests addObject:copiedRequest];
+    }
+    @synchronized(self.requestHandlers) {
```

Comment:
Did you consider trying to put the handler block on the request itself in some way (like userInfo).  There would be pros and cons to that, for example, it might be a little weird to modify the request in that way and have the blocks existence leak out of the response object, but I am curious if that can would be outweighed by not having to use synchronized.

Reply:
I believe we discussed this internally and decided not to attach completion blocks to requests in the userInfo field.